config.go

```diff
@@ -205,6 +219,18 @@ func (cfg Config) buildOptions(errSink zapcore.WriteSyncer) []Option {
 
 	if cfg.Sampling != nil {
 		opts = append(opts, WrapCore(func(core zapcore.Core) zapcore.Core {
+			if cfg.Sampling.Reporting != nil && cfg.Sampling.Reporting.Enabled {
+				loggerName := cfg.Sampling.Reporting.LoggerName
+				if loggerName == "" {
+					loggerName = "zapcore"
+				}
+				level := cfg.Sampling.Reporting.Level.Level()
+				message := cfg.Sampling.Reporting.Message
+				if message == "" {
```

Comment:
Push this zero default down into the `NewReportingSampler` constructor; likewise for the loggerName above.